src/logplex_tlssyslog_drain.erl

```diff
@@ -423,20 +447,56 @@ connect(#state{sock = undefined, host=Host, port=Port})
                 T when is_tuple(T) -> T;
                 A when is_atom(A) -> A
             end,
-    Options = [binary
+
+    CertOpts = compile_cert_opts([verify, depth, cacertfile]),
+    ?INFO("drain_id=~p channel_id=~p dest=~s "
+          "cert_opts=~p",
+          log_info(State, [CertOpts])),
+
+    Options = CertOpts ++ [binary
                %% We don't expect data, but why not.
                ,{active, true}
                ,{exit_on_close, true}
                ,{keepalive, true}
                ,{packet, raw}
                ,{reuseaddr, true}
-               %% XXX - put ssl connection (cert validaton) options here
               ],
     ssl:connect(HostS, Port, Options,
                 timer:seconds(SendTimeoutS));
 connect(#state{}) ->
     {error, bogus_port_number}.
 
+compile_cert_opts(OptList) ->
```

Comment:
This whole lot of functions probably should be in a separate tls_client utility function module. You'll need the functions again when it comes to configuring https validation, any https calls made from other places too - this is widely applicable stuff.

(Arguably, setting up Erlang SSL clients correctly for cert validation should live in its on OTP library so it can be included in all our Erlang apps).